### PR TITLE
cmake-language-server: 0.1.2 → 0.1.3

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Motivation for this change
Update to the latest version: https://github.com/regen100/cmake-language-server/releases/tag/v0.1.3 and also update pygls as a requirement: https://github.com/openlawlibrary/pygls/releases/tag/v0.11.2.

This change also fixes the test failure introduced when python3 was updated to v3.9: https://hydra.nixos.org/build/154473169.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 140440` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>cmake-language-server</li>
    <li>python38Packages.pygls</li>
    <li>python39Packages.pygls</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 140440` at 72478046 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>cmake-language-server</li>
    <li>python38Packages.pygls</li>
    <li>python39Packages.pygls</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/tools/cmake-language-server/default.nix:26:5:
```
   |
26 |     ./disable-test-timeouts.patch
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/development/tools/cmake-language-server/default.nix:2:3:
```
  |
2 | , stdenv
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pygls/default.nix [hunk 1]
```diff
@@ -2,24 +2,41 @@
...
-  version = "0.9.1";
+  version = "0.11.2";
   disabled = !isPy3k;
 
   src = fetchFromGitHub {
     owner = "openlawlibrary";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1v7x5598d6jg8ya0spqjma56y062rznwimsrp8nq6fkskqgfm0ds";
+    hash = "sha256-zgQ5m198HMyFFrASSYCzn0EDLLeVy2j4LD0rEyEgahQ=";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pygls/default.nix:22]:
```suggestion
    sha256 = "sha256-zgQ5m198HMyFFrASSYCzn0EDLLeVy2j4LD0rEyEgahQ=";
```

File: pkgs/development/tools/cmake-language-server/default.nix [hunk 2]
```diff
@@ -1,46 +1,41 @@
...
-  version = "0.1.2";
+  version = "0.1.3";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "regen100";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0vz7bjxkk0phjhz3h9kj6yr7wnk3g7lqmkqraa0kw12mzcfck837";
+    hash = "sha256-eZBnygEYjLzk29tvLGg1JdhCECc5x2MewHRSChMuCjo=";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/cmake-language-server/default.nix:21]:
```suggestion
    sha256  = "sha256-eZBnygEYjLzk29tvLGg1JdhCECc5x2MewHRSChMuCjo=";
```
just stick to sha256 for now. It makes future updates easier and just works with everything.

Comment (kira-bruneau) [reply pkgs/development/tools/cmake-language-server/default.nix:21]: Ok, that makes sense! Most of my packages use `hash` right now, but I'll slowly trickle them back to `sha256` as they get updated.

**Pull request merged**